### ISSUE TITLE
Some tasks are stuck in queued state

### ISSUE THREAD

**Issue by bluzy** [apache/airflow#32127]
### Apache Airflow version

Other Airflow 2 version (please specify below)

### What happened

We are using airflow 2.2.4 with CeleryExecutor.

The problem is that a task is stuck in queued state sometimes.
We are using same codes in DEV and PROD environment, but the problem is occurred on only DEV environment.
And stuck tasks are different for each case, but same level of dag - we are creating tasks in a loop.

I couldn't get a hint from log,
```
[2023-06-20 10:15:24,936] {scheduler_job.py:473} INFO - Sending TaskInstanceKey(dag_id='hourly_iceberg_snapshot', task_id='pup_snapshot.check_media-item_maintenance_time', run_id='scheduled__2023-06-19T18:00:00+00:00', try_number=1) to executor with priority 3 and queue default
[2023-06-20 10:15:24,936] {base_executor.py:85} ERROR - could not queue task TaskInstanceKey(dag_id='hourly_iceberg_snapshot', task_id='pup_snapshot.check_media-item_maintenance_time', run_id='scheduled__2023-06-19T18:00:00+00:00', try_number=1)
```
Is there a point to fix it?

### How to reproduce

I don't know how to reproduce manually, it seems to be happened randomly.

### Operating System

centos7

### Versions of Apache Airflow Providers

apache-airflow-providers-apache-hive==2.2.0
apache-airflow-providers-apache-spark==2.1.0
apache-airflow-providers-celery==2.1.0
apache-airflow-providers-cncf-kubernetes==3.0.2
apache-airflow-providers-ftp==2.0.1
apache-airflow-providers-http==2.0.3
apache-airflow-providers-imap==2.2.0
apache-airflow-providers-jdbc==2.1.0
apache-airflow-providers-postgres==3.0.0
apache-airflow-providers-redis==2.0.1
apache-airflow-providers-sqlite==2.1.0
apache-airflow-providers-ssh==2.4.0

### Deployment

Official Apache Airflow Helm Chart

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by eladkal** [comment]
Please update Airflow version.
Airflow 2.6 has significantly improved this part of the code.
See [Airflow 2.6.0 release notes ](https://airflow.apache.org/docs/apache-airflow/2.6.0/release_notes.html#consolidate-handling-of-tasks-stuck-in-queued-under-new-task-queued-timeout-config-30375)

**Comment by potiuk** [comment]
And the https://medium.com/apache-airflow/unsticking-airflow-stuck-queued-tasks-are-no-more-in-2-6-0-6f40a1a22835 blog post as well. Closing for now, until the user upgrades, we can reopen if the problem is not fixed in 2.6.* (BTW. We of course recomment to upgrade to latest 2.6 version released - which at the moment of this writing is 2.6.2)
